drgn/helpers/linux/slab.py

```diff
@@ -92,6 +105,66 @@ def slab_cache_is_merged(slab_cache: Object) -> bool:
     return slab_cache.refcount > 1
 
 
+def get_slab_cache_aliases(prog: Program) -> Dict[str, str]:
+    """
+    Return a dict mapping slab cache name to the cache it was merged with.
+
+    The SLAB and SLUB subsystems can merge caches with similar settings and
+    object sizes, as described in the documentation of
+    :func:`slab_cache_is_merged()`. In some cases, the information about which
+    caches were merged is lost, but in other cases, we can reconstruct the info.
+    This function reconstructs the mapping, but requires that the kernel is
+    configured with ``CONFIG_SLUB`` and ``CONFIG_SYSFS``.
+
+    The returned dict maps from original cache name, to merged cache name. You
+    can use this mapping to discover the correct cache to lookup via
+    :func:`find_slab_cache()`. The dict contains an entry only for caches which
+    were merged into a cache of a different name.
+
+    >>> cache_to_merged = get_slab_cache_aliases(prog)
+    >>> cache_to_merged["dnotify_struct"]
+    'avc_xperms_data'
+    >>> "avc_xperms_data" in cache_to_merged
+    False
+    >>> find_slab_cache(prog, "dnotify_struct") is None
+    True
+    >>> find_slab_cache(prog, "avc_xperms_data") is None
+    False
+
+    :warning: This function will only work on kernels which are built with
+      ``CONFIG_SLUB`` and ``CONFIG_SYSFS`` enabled.
+
+    :param prog: Program to search
+    :returns: Mapping of slab cache name to final merged name
+    """
+    try:
+        slab_kset = prog["slab_kset"]
+    except KeyError:
+        raise ValueError(
```

Comment:
The `get_kconfig()` helper uses `LookupError` for a similar case: https://github.com/osandov/drgn/blob/72f1db796844c36ef7cf542166ad44336881e19e/drgn/helpers/linux/kconfig.py#L49-L51

Do you think that would make more sense here, too?

Either way, this should be `raise foo from None` so that we don't get extra noise from the original `KeyError`; I just fixed this for `get_kconfig()`, too: https://github.com/osandov/drgn/commit/72f1db796844c36ef7cf542166ad44336881e19e.

Reply:
LookupError is fine by me - I'll note it in the documentation too, since the whole point here seems to be that we want users to be able to know _which_ exception to catch if their kernel doesn't support the helper.